webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala

```diff
@@ -893,14 +893,10 @@ class ComplianceAPIService(
       allDirectives:         Map[DirectiveId, (FullActiveTechnique, Directive)],
       nodeFacts:             MapView[NodeId, CoreNodeFact],
       nodeSettings:          Map[NodeId, RudderSettings],
-      rules:                 Seq[Rule],
-      allRuleInfos:          Map[RuleId, (Chunk[NodeId], Option[PolicyMode])],
+      rules:                 Map[RuleId, (Rule, Chunk[NodeId], Option[PolicyMode])],
       level:                 Option[Int],
       isGlobalCompliance:    Boolean
   )(implicit qc: QueryContext): IOResult[ByNodeGroupCompliance] = {
-    val ruleMap = rules.map(r => (r.id, r)).toMap
-    val ruleIds = ruleMap.keySet
```

Comment:
I think it's not really necessary. It is not from a perf point of view, since keySet is built when the map is built (and is actually a false keySet, it forward operation to the map, which likely mean the map can't be gc'ed if we pass the set around, but that has nothing to do with the story here)